src/ptn.rs

```diff
@@ -40,6 +40,46 @@ impl Square {
     pub fn row(self) -> u8 {
         self.row
     }
+
+    const fn assert_on_board(self, board_size: u8) {
+        assert!(self.row < board_size);
+        assert!(self.column < board_size);
+    }
+
+    /// Generate a Square that is one step in the direction from this Square.
+    /// If the generated Square is outside of the board, then this returns None instead.
+    pub fn checked_step(self, direction: Direction, board_size: u8) -> Option<Self> {
+        self.assert_on_board(board_size);
+        use Direction::*;
+        let (column, row) = (self.column, self.row);
+        let (column, row) = match direction {
+            Up => (column, row.checked_add(1).filter(|&y| y < board_size)?),
+            Down => (column, row.checked_sub(1)?),
+            Left => (column.checked_sub(1)?, row),
+            Right => (column.checked_add(1).filter(|&x| x < board_size)?, row),
+        };
+        Some(Self { column, row })
+    }
+
+    /// Rotate 1 quarter turn counterclockwise.
```

Comment:
Docs disagree with tests. Also, it would be inconsistent to rotate Squares clockwise and Directions counterclockwise.